### PR TITLE
Test Runner Implementation

### PR DESCRIPTION
**Pull request by Tan108**

### CODE REVIEW

File: src/main/kotlin/com/featurevisor/cli/AssertionHelper.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package com.featurevisor.cli
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/cli/AssertionHelper.kt:1]: given we are no longer exposing a `featurevisor-kotlin` command, and instead expecting to use the Test Runner via  Gradle task, I think it makes sense now to rename `cli` directory to `testRunner`.

for comparison, `featurevisor-swift` did the same: https://github.com/featurevisor/featurevisor-swift/tree/main/Sources/FeaturevisorTestRunner

File: src/main/kotlin/com/featurevisor/cli/TestExecuter.kt [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        getJsonForFeatureUsingCommand(
+            featureName = featureName,
+            environment = environment,
+            projectRootPath = projectRootPath
+        )?.run {
+            convertToDataClass<DatafileContent>()
+        }
+    } catch (e: Exception) {
+        printMessageInRedColor("Exception while parsing data file --> ${e.message}")
+        null
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/cli/TestExecuter.kt:179]: is this line needed? or should we `return` it?

File: src/main/kotlin/com/featurevisor/cli/TestFeature.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                                         expected = expectedValueString,
+                                         actual = actualValueString,
+                                         details = mapOf("variableKey" to variableKey)
+                                     )
+                                 )
+                             }
+                         }
+                     }
+                 }
+             }else{
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/cli/TestFeature.kt:181]: is there any formatting tool for Kotlin, that can help enforce having spaces around `else`?

or is it more idiomatic in Kotlin to write it this way as above?

File: src/main/kotlin/com/featurevisor/sdk/serializers/Serializers.kt [hunk 4]
```diff
@@ -34,6 +35,34 @@ import kotlinx.serialization.json.jsonObject
...
+            is JsonArray -> {
+                Required.FeatureKey(tree.toString())
+            }
+
+            else -> Required.FeatureKey("abc")
+        }
+    }
+}
+
+
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/sdk/serializers/Serializers.kt:62]: may be some code formatting tool can help us remove redundant newlines?

File: src/main/kotlin/com/featurevisor/types/Types.kt [hunk 5]
```diff
@@ -388,3 +344,98 @@ data class OverrideFeature(
     val variation: VariationValue? = null,
     val variables: VariableValues? = null,
 )
+
+/**
+ * Tests
+ */
+
+typealias AssertionMatrix = Map<String, List<AttributeValue>>
```
Comment (fahad19) [inline src/main/kotlin/com/featurevisor/types/Types.kt:352]: I wouldn't mind seeing testing related types ending up in the `types` directory. 

we are doing similar with the TypeScript package.

Comment (fahad19) [reply src/main/kotlin/com/featurevisor/types/Types.kt:352]: I am so sorry. ignore my comment. I see this is the common types file.

File: src/main/kotlin/com/featurevisor/cli/AssertionHelper.kt (hunk 1 above)
Comment (Tan108) [reply src/main/kotlin/com/featurevisor/cli/AssertionHelper.kt]: Done -  renamed to testRunner

**Review by harshit-sinha-dazn — APPROVED** [review]

**Pull request merged**